### PR TITLE
new keyboard tata80

### PR DESCRIPTION
**Pull request by spbgzh**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/wuque/tata80/wk/wk.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        	DDRE &= ~(1 << 6); PORTE &= ~(1 << 6);
+        }
+        else
+        {
+        	DDRE |= (1 << 6); PORTE|= (1 << 6);
+        }
+    }
+    return res;
+}
+
```
Comment (waffle87) [inline keyboards/wuque/tata80/wk/wk.c:43]:
```suggestion
```

File: keyboards/wuque/tata80/wk/info.json [hunk 2]
```diff
@@ -0,0 +1,110 @@
+{
+  "keyboard_name": "Tata80_wk",
+  "manufacturer": "wuque studio",
+  "url": "https://shop.wuquestudio.com/products/tata80-keyboard-kit",
+  "maintainer": "Koichi Katano",
+  "usb": {
+    "vid": "0x2365",
+    "pid": "0x0168",
+    "device_version": "0.0.1"
+  },
```
Comment (waffle87) [inline keyboards/wuque/tata80/wk/info.json:10]:
```suggestion
  },
  "indicators": {
      "caps_lock": "E6"
  },
```

File: keyboards/wuque/tata80/wkl/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+# SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+VIA_ENABLE = yes            # 2022-09-02
```
Comment (waffle87) [inline keyboards/wuque/tata80/wkl/rules.mk:22]:
```suggestion
```

File: keyboards/wuque/tata80/wkl/wkl.c [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        	DDRE &= ~(1 << 6); PORTE &= ~(1 << 6);
+        }
+        else
+        {
+        	DDRE |= (1 << 6); PORTE|= (1 << 6);
+        }
+    }
+    return res;
+}
+
```
Comment (waffle87) [inline keyboards/wuque/tata80/wkl/wkl.c:43]:
```suggestion
```

File: keyboards/wuque/tata80/wkl/info.json [hunk 5]
```diff
@@ -0,0 +1,109 @@
+{
+  "keyboard_name": "Tata80_wkl",
+  "manufacturer": "wuque studio",
+  "url": "https://shop.wuquestudio.com/products/tata80-keyboard-kit",
+  "maintainer": "Koichi Katano",
+  "usb": {
+    "vid": "0x2365",
+    "pid": "0x0169",
+    "device_version": "0.0.1"
+  },
```
Comment (waffle87) [inline keyboards/wuque/tata80/wkl/info.json:10]:
```suggestion
  },
  "indicators": {
      "caps_lock": "E6"
  },
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/wuque/tata80/wk/rules.mk [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+#
+BOOTMAGIC_ENABLE = yes       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+# SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/wuque/tata80/wk/rules.mk:18]: Please remove the `SLEEP_LED_ENABLE` setting, as it's not really used, and remove the comments for both it and NKRO.  
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/wuque/tata80/wkl/rules.mk [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+#
+BOOTMAGIC_ENABLE = yes       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+# SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/wuque/tata80/wkl/rules.mk:18]: Please remove the `SLEEP_LED_ENABLE` setting, as it's not really used, and remove the comments for both it and NKRO.  
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by drashna — APPROVED** [review]

File: keyboards/wuque/tata80/wkl/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,            KC_PSCR, KC_SLCT, KC_PAUS,
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wkl/keymaps/via/keymap.c:20]:
```suggestion
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,            KC_PSCR, KC_SCRL, KC_PAUS,
```

File: keyboards/wuque/tata80/wkl/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,          KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),   KC_APPLICATION,   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wkl/keymaps/via/keymap.c:25]:
```suggestion
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),   KC_APP,   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```

File: keyboards/wuque/tata80/wkl/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,           KC_PSCR, KC_SLCT, KC_PAUS,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,          KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),   KC_APPLICATION,   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wkl/keymaps/default/keymap.c:26]:
```suggestion
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),   KC_APP,   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```

File: keyboards/wuque/tata80/wkl/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,           KC_PSCR, KC_SLCT, KC_PAUS,
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wkl/keymaps/default/keymap.c:21]:
```suggestion
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,           KC_PSCR, KC_SCRL, KC_PAUS,
```

File: keyboards/wuque/tata80/wk/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+       [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_F13,  KC_PSCR, KC_SLCT, KC_PAUS,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,          KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),   KC_APPLICATION,   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wk/keymaps/via/keymap.c:25]:
```suggestion
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),   KC_APP,   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```

File: keyboards/wuque/tata80/wk/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+       [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_F13,  KC_PSCR, KC_SLCT, KC_PAUS,
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wk/keymaps/via/keymap.c:20]:
```suggestion
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_F13,  KC_PSCR, KC_SCRL, KC_PAUS,
```

File: keyboards/wuque/tata80/wk/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_F13,  KC_PSCR, KC_SLCT, KC_PAUS,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,          KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),   KC_APPLICATION,   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wk/keymaps/default/keymap.c:26]:
```suggestion
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),   KC_APP,   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```

File: keyboards/wuque/tata80/wk/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_F13,  KC_PSCR, KC_SLCT, KC_PAUS,
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wk/keymaps/default/keymap.c:21]:
```suggestion
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_F13,  KC_PSCR, KC_SCRL, KC_PAUS,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by spbgzh — COMMENTED** [review]
fix

File: keyboards/wuque/tata80/wkl/info.json [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wkl/info.json:10]:
```suggestion
  },
  "processor": "atmega32u4",
  "bootloader": "atmel-dfu",
```

File: keyboards/wuque/tata80/wkl/rules.mk [hunk 17]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wkl/rules.mk:6]:
```suggestion
```

File: keyboards/wuque/tata80/wk/rules.mk [hunk 18]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wk/rules.mk:6]:
```suggestion
```

File: keyboards/wuque/tata80/wk/info.json [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wk/info.json:10]:
```suggestion
  },
  "processor": "atmega32u4",
  "bootloader": "atmel-dfu",
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/wuque/tata80/wk/config.h [hunk 20]
```diff
@@ -0,0 +1,61 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "config_common.h"
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 17
+
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wk/config.h:23]:
```suggestion
```

File: keyboards/wuque/tata80/wk/keymaps/via/rule.mk [hunk 21]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wk/keymaps/via/rule.mk:1]: This file needs to be named `rules.mk`.

File: keyboards/wuque/tata80/wkl/config.h [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "config_common.h"
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 17
+
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wkl/config.h:23]:
```suggestion
```

File: keyboards/wuque/tata80/wkl/keymaps/via/rule.mk [hunk 23]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/wuque/tata80/wkl/keymaps/via/rule.mk:1]: As above.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**